scripts/hubspot-deploy/hubspot-api.js

```diff
@@ -1,100 +1,119 @@
-const axios = require('axios');
+const axios = require("axios");
 const Bottleneck = require("bottleneck");
 
 var rawStart = "{% raw %}",
-    rawEnd = "{% endraw %}";
+  rawEnd = "{% endraw %}";
 
 class HubSpotApi {
-  constructor(apiKey, campaignId){
-    this.apiKey = apiKey;
+  constructor(accessToken, campaignId) {
+    this.accessToken = accessToken;
     this.campaignId = campaignId;
-    this.baseUrl = 'https://api.hubapi.com/content/api/v2/pages';
-    this.limiter = new Bottleneck({minTime: 150})
+    this.baseUrl = "https://api.hubapi.com/content/api/v2/pages";
+    this.limiter = new Bottleneck({ minTime: 150 });
+    this.axios = axios.create({
+      baseURL: this.baseUrl,
+      headers: {
+        Authorization: `Bearer ${this.accessToken}`,
+        "Content-Type": "application/json",
+      },
+    });
   }
 
-  makeRequest(method, url, data){
-    return this.limiter.schedule(() => axios({ method, url, data })
-      .catch(error => {
-        if (data.name) {
-          console.error(`Error on page ${data.name}`);
-        }
+  makeRequest(method, url, data) {
+    console.log("Making request:", method, url, data);
```

Comment:
This method is commented out internally? Do you need to delete it? It's used in multiple places.

Reply:
I commented it out on advice from Christopher so that when we merge this into main it just logs to the console instead of touching the site.

Reply:
Ok, so this let's you test run it locally, but when you actually want to update the site you uncomment the rest of the lines.